### PR TITLE
Nomad Vault Integration - Ansible

### PR DESCRIPTION
**Pull request by phanclan**
Created two new nomad config files that will define nomad vault integration.
Changed playbook to run Vault tasks before Nomad since Nomad now depends on Vault.

### CODE REVIEW

File: ansible/roles/nomad/templates/server_vault.hcl.j2 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#
+# // Vault Integration
+#
+
+vault {
+  enabled = true
+  address = "http://{{vault_server}}:8200"
+  #token   = "REPLACETOKEN"
```
Comment (rberlind) [inline ansible/roles/nomad/templates/server_vault.hcl.j2:8]: I know the token is commented out, but why not delete it entirely?

File: ansible/roles/nomad/tasks/main.yml [hunk 2]
```diff
@@ -116,11 +118,22 @@
...
+    - client_vault
   notify: nomad_restart
   tags:
     - nomad
   when: "'client' in inventory_hostname"
 
+- name: Insert Vault token into nomad vault integration agent config.
+  replace:
+    path: /etc/nomad.d/server_vault.hcl #"{{ item.path }}"
+    regexp: "REPLACETOKEN"
```
Comment (rberlind) [inline ansible/roles/nomad/tasks/main.yml:130]: I don't really know Ansible, but I would think we would no longer need this given the plan to use environment variable VAULT_TOKEN

**Review by rberlind — COMMENTED** [review]
@phanclan : I would think we would be getting rid of REPLACETOKEN altogether.

File: ansible/roles/nomad/templates/server_vault.hcl.j2 (hunk 1 above)
Comment (phanclan) [reply ansible/roles/nomad/templates/server_vault.hcl.j2:8]: in case it doesn't work i can revert

File: ansible/roles/nomad/tasks/main.yml (hunk 2 above)
Comment (phanclan) [reply ansible/roles/nomad/tasks/main.yml:130]: Once we validate things are working I will remove.

**Comment by rberlind** [comment]
Thanks @phanclan

**Pull request merged**